services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java

```diff
@@ -113,4 +348,34 @@ public void listActions(CallContext callContext, StreamListener<ActionType> stre
   private CallHeaders retrieveHeadersFromCallContext(CallContext callContext) {
     return callContext.getMiddleware(FlightConstants.HEADER_KEY).headers();
   }
+
+  private boolean isFlightSqlCommand(Any command) {
+    return command.is(CommandStatementQuery.class) || command.is(CommandPreparedStatementQuery.class) ||
+      command.is(CommandGetCatalogs.class) || command.is(CommandGetSchemas.class) ||
+      command.is(CommandGetTables.class) || command.is(CommandGetTableTypes.class) ||
+      command.is(CommandGetSqlInfo.class) || command.is(CommandGetPrimaryKeys.class) ||
+      command.is(CommandGetExportedKeys.class) || command.is(CommandGetImportedKeys.class);
+  }
+
+  private boolean isFlightSqlCommand(byte[] bytes) {
+    try {
+      Any command = Any.parseFrom(bytes);
+      return isFlightSqlCommand(command);
+    } catch (InvalidProtocolBufferException e) {
+      return false;
+    }
+  }
+
+  private boolean isFlightSqlCommand(FlightDescriptor flightDescriptor) {
+    return isFlightSqlCommand(flightDescriptor.getCommand());
+  }
+
+  private boolean isFlightSqlCommand(Ticket ticket) {
```

Comment:
I don't think I follow this. When can a ticket ever be a FlightSqlCommand?

Reply:
Fixed - actually the ticket bytes are a serialized FlightSqlCommand, renamed and added comments